### DISCUSSION TITLE
Multiple root element check fails after web performance optimization strips comments in HTML

### DISCUSSION THREAD

**Discussion by hein-p** [livewire/livewire#4015] [7. Bugs (archived)]
The check for multiple root elements in JavaScript (function `checkForMultipleRootElements()`) is dependent on a comment in the HTML indicating the end of the Livewire component. For example:

```html
<div>
	<div wire:id="WPed52ADewWZbPHtf93p">
		...
	</div>
	<!-- Livewire Component wire-end:WPed52ADewWZbPHtf93p --> 
	<div>
		...
	</div>
</div>
```

However, if your website is being optimized for example with [Cloudflare's Auto Minify](https://www.cloudflare.com/website-optimization/) then HTML comments are removed from the HTML being served. The `checkForMultipleRootElements`  function then does not work as expected when the Livewire component is not the last element in the parent, and incorrectly throws the multiple root elements warning.

**Comment by joshhanley** [comment]
@hein-p thanks for reporting, I think that is the same issue as reported in #3887.

I haven't had a chance to see if I can fix it yet, but it shouldn't impact the functioning of the app, as they are just console warnings.

When I get some time though I'll see if we can get it fixed.

Hope this helps!

**Reply by hein-p** [reply]
@joshhanley since this is not a critical issue, I was wondering if you mind if I try to fix the issue? I have been meaning to find a open source project to contribute to.

**Reply by joshhanley** [reply]
@hein-p yeah go for it! All contributions are welcome and encouraged! 🙂

Fair warning, I don't think this is going to be a straight forward issue, although I'm more thinking about how to stop the comment from being removed in a parent/ child scenario as listed in #3887.

But if you run into any problems, throw your questions up on this discussion and I'll help when I can 🙂

Thanks!

**Reply by hein-p** [reply]
@joshhanley I opened a PR to fix #3887 (https://github.com/livewire/livewire/pull/4153). I doubt it would do anything about this issue though.

**Reply by joshhanley** [reply]
@hein-p thanks! Yeah, may have to change to using template tags instead of comments maybe

**Reply by hpieterse** [reply]
@joshhanley I googled a bit, and turns out Cloudflare does not remove conditional comments.

https://community.cloudflare.com/t/omit-formatted-comments-from-minification/18572/25
https://en.wikipedia.org/wiki/Conditional_comment

I added a small change to the PR (#4153) to also fix this issue.
